### PR TITLE
Replaced internal quote cache with PostgreSQL database.

### LINKED ISSUE #80: Username changes result in previous quotes not being accessible
**Describe the bug**
Nickname changes cause the quotes function to not retrieve all quotes

**To Reproduce**
1. Quote a message
2. Change nickname
3. View quotes

**Expected behavior**
All quotes shown. Instead, none shown.

**Additional context**
Could be detected via discriminator, but Nitro users would break that as well.

### PR DESCRIPTION
**Pull request by fractalpenguin**
- Replaced internal cache with a database: simply run the `migrate_quotes` command. Needs the `PG_HOST`, `PG_USER`, `PG_DATABASE` and `PG_PASSWORD` environmental variables to be set.
- Added the required asyncpg dependency to both `Pipfile` and `requirements.txt`
- Resolves #80.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -13,12 +13,17 @@ flake8-tidy-imports = "==1.1.0"
 flake8-todo = "==0.7"
 safety = "==1.8.4"
 dodgy = "==0.1.9"
+python-language-server = {extras = ["all"]}
```
Comment (AnonGuy) [inline Pipfile]: delet this

Comment (fractalpenguin) [reply Pipfile]: ok

**Comment by fractalpenguin** [comment]
This PR will fix #80.

**Comment by fractalpenguin** [comment]
This will fix #80 .

File: bot/cogs/fun.py [hunk 2]
```diff
@@ -193,24 +206,87 @@ def __init__(self, bot: Bot):
...
+            )
+            if message_id is None:
+                await ctx.send("No quotes for that user.")
                 return
-            message_id = choice(user_quotes)
 
+        await conn.close()
         message = await quote_channel.get_message(message_id)
-        await ctx.send(embed=message.embeds[0])
+        if message.author.id == QUOTES_BOT_ID:
```
Comment (shadycake) [inline bot/cogs/fun.py]: Line 227 to 235 don't account for all possible types of quote, here's a suggestion but I'm not a perfect programmer so feel free to chip in (This should account for embeds and multiple attachments and embedding attachments where possible)
```suggestion
        embed = None
        content = message.content
        attachment_urls = [attachment.url for attachment in message.attachments]

        if message.embeds: 
            embed = message.embeds[0]
        elif len(attachments) == 1:
            image_url = attachments.pop(0)
            embed = Embed()
            embed.set_image(url=image_url)

        for url in attachments_urls:
            content += "\n" + url

        await ctx.send(content, embed=embed)
```

File: bot/cogs/fun.py [hunk 3]
```diff
@@ -193,24 +206,87 @@ def __init__(self, bot: Bot):
...
+            quote = Embed()
+            quote.set_image(url=image_url)
+            await ctx.send(embed=quote)
+        else:
+            await ctx.send(message.content)
+
+    async def add_quote_to_db(
+        self, conn: asyncpg.connection.Connection, quote: Message
+    ):
+        if quote.author.id == QUOTES_BOT_ID:
```
Comment (shadycake) [inline bot/cogs/fun.py]: I'd suggest commenting this bit, since it's not entirely clear what's going on

**Comment by thebeanogamer** [comment]
This is... wow!
image

**Comment by thebeanogamer** [comment]
In addition, there are now merge conflicts that need to be resolved.

**Comment by AnonGuy** [comment]
Fixed merge conflicts.

File: bot/cogs/fun.py (hunk 3 above)
Comment (fractalpenguin) [reply bot/cogs/fun.py]: done.

**Comment by fractalpenguin** [comment]
Beano.exe has stopped working

**Pull request merged**